src/foam/dao/AbstractJDAO.java

```diff
@@ -58,16 +58,24 @@ public AbstractJDAO(foam.core.X x, DAO delegate, String filename) {
     }
 
     logger_ = new PrefixLogger(new Object[] { "[JDAO]", filename }, logger);
-
     try {
-      file_ = getX().get(foam.nanos.fs.Storage.class).get(filename);
-
-      if ( ! file_.exists() ) file_.createNewFile();
-
-      setDelegate(delegate);
-      loadJournal();
-
-      out_ = new BufferedWriter(new FileWriter(file_, true));
+      //get repo entries in filename.0 journal first
+      File infile = getX().get(foam.nanos.fs.Storage.class).get(filename + ".0");
```

Comment:
Inconsistent naming. You have "infile" and "writeFile", which mix camel-case and non-camel-case with in/out vs read/write.